### PR TITLE
Khn 46

### PR DESCRIPTION
**Pull request by Tarasovas2212**
Base classes are finished and CRUD both for users and cards is created.

### CODE REVIEW

File: src/controllers/CardController.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+const { cardService } = require('../services');
+const MainController = require('./MainController');
+
+class CardController extends MainController {}
+module.exports = new CardController(cardService);
```
Comment (asavienko) [inline src/controllers/CardController.js:5]: use prettier 
add a blank line before and after module.exports

File: src/routes/cards.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+const express = require('express');
+const { cardController } = require('../controllers');
+
+const router = express.Router();
+
+router.get('/', cardController.getAll.bind(cardController));
```
Comment (asavienko) [inline src/routes/cards.js]:
```suggestion
router.get('/', (req, res, next)=> cardController.getAll(req, res, next) );
```
so you can use arrow function heir

File: src/controllers/MainController.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  async get({ params: { id } = {} }, res) {
+    const response = await this.service.get(id);
+    res.json(response);
+    res.sendStatus(response ? 200 : 404);
+  }
+
+  async create(req, res) {
+    const user = await this.service.create(req.body);
+    res.json(user);
+  }
```
Comment (asavienko) [inline src/controllers/MainController.js]:
```suggestion
  async create({body}, res) {
    const user = await this.service.create(body);
    res.json(user);
  }
```

use destruction

File: src/routes/index.js [hunk 4]
```diff
@@ -1,8 +1,10 @@
 const express = require('express');
 const users = require('./users');
+const cards = require('./cards');
 
 const router = express.Router();
 
 router.use('/users/', users);
+router.use('/cards/', cards);
```
Comment (asavienko) [inline src/routes/index.js]:
```suggestion
router.use('/users', users);
router.use('/cards', cards);
```

keep project style

File: src/services/CardService.js [hunk 5]
```diff
@@ -0,0 +1,5 @@
+const { card } = require('../models');
+const MainService = require('./MainService');
+
+class CardService extends MainService {}
+module.exports = new CardService(card);
```
Comment (asavienko) [inline src/services/CardService.js:5]: the same as above

**Review by asavienko — CHANGES_REQUESTED** [review]

File: .gitignore [hunk 6]
```diff
@@ -1,2 +1,3 @@
 /.env
 node_modules
+vscode
```
Comment (vlad-yelnikov) [inline .gitignore]:
```suggestion
.vscode
```

**Pull request merged**